block/block.go

```diff
@@ -17,9 +17,15 @@ import (
 // In case the following doesn't hold true, it means we crashed after the commit and before updating the store height.
 // In that case we'll want to align the store with the app state and continue to the next block.
 func (m *Manager) applyBlock(ctx context.Context, block *types.Block, commit *types.Commit, blockMetaData blockMetaData) error {
-	if block.Header.Height != m.store.Height()+1 {
+	//validate block height
+	if m.lastState.IsGenesis() {
```

Comment:
as this check will be done every time and it's likely only relevant once, I think it's better to move it to another place. Also not sure whats the scenario we're checking exactly. comments would be helpful.

Reply:
when u apply block, u expects it's height to be the expected height.
the expected height is different between genesis and post genesis